src/engine/SCons/Script/Main.py

```diff
@@ -1099,17 +1100,36 @@ def _main(parser):
                                           targets, target_top)
 
     else:
-        # Write the build graph
-        graph = os.environ.get("SCONS_GRAPH")
-        if graph:
+        scons_mode = os.environ.get("SCONS_MODE")
+        if options.mode == "vulcan" or scons_mode == "vulcan":
+            # Write the build graph
+            graph = "scons.gv"
```

Comment:
You probably want to use a tempfile for this instead of writing to the current working directory.